File: easy/Halloween_Sale.go
package main
import "fmt"
func prices(price, diff, min, stoc int32 ) int32 {
    var result int32
    for stoc >= price {
        result++
        stoc  = stoc - price
        price -= diff
        if price < min {
            price = min
        }
    }
    return result
}

func main() {
    r := prices(20, 3, 6, 85)
    fmt.Println("+++++++++++", r)
}

File: easy/Halloween_Sale.lua
function howManyGames(price, diff, min, stoc)
    -- Return the number of games you can buy
    result = 0
    var = 0
    while stoc >= price do
        result = result + 1
        stoc  = stoc - price
        price = price-diff
        if price < min then
            price = min
        end
    end
    return result

end

r = howManyGames(20, 3, 6, 85)
print(r)
